niworkflows/viz/plots.py

```diff
@@ -1051,6 +1051,104 @@ def confounds_correlation_plot(
     return [ax0, ax1], gs
 
 
+def cifti_surfaces_plot(
+    in_cifti,
+    density="32k",
+    surface_type="inflated",
+    output_file=None,
+    clip_range=None,
+    **splt_kwargs
+):
+    """
+    Plots a CIFTI-2 dense timeseries onto left/right mesh surfaces.
+
+    Parameters
+    ----------
+    in_cifti : str
+        CIFTI-2 dense timeseries (.dtseries.nii)
+    density : str
+        Surface density
+    surface_type : str
+        Inflation level of mesh surfaces. Supported: midthickness, inflated, veryinflated
+    clip_range : None or tuple
+        Range to clip `in_cifti` data prior to plotting.
+        If not None, two values must be provided as lower and upper bounds.
+        If values are None, no clipping is performed for that bound.
+    output_file: :obj:`str` or :obj:`None`
+        Path where the output figure should be saved. If this is not defined,
+        then the figure will be returned.
+    splt_kwargs : dict
+        Keyword arguments for :obj:`surfplot.Plot`
+
+    Outputs
+    -------
+    figure : matplotlib.pyplot.figure
+        Surface plot figure. Returned only if ``output_file`` is ``None``.
+    output_file: :obj:`str`
+        The file where the figure is saved.
+    """
+    import surfplot as splt
+    from surfplot.utils import add_fslr_medial_wall
+
+    def get_surface_meshes(density, surface_type):
+        import templateflow.api as tf
+        lh, rh = tf.get("fsLR", density=density, suffix=surface_type, extension=[".surf.gii"])
+        return str(lh), str(rh)
+
+    if density != "32k":
+        raise NotImplementedError("Only 32k density is currently supported.")
+
+    img = nb.cifti2.load(in_cifti)
+    if img.nifti_header.get_intent()[0] != "ConnDenseSeries":
+        raise TypeError(f"{in_cifti} is not a dense timeseries CIFTI file")
+
+    geo = img.header.get_index_map(1)
+    left_cortex, right_cortex = None, None
+    for bm in geo.brain_models:
+        if bm.brain_structure == "CIFTI_STRUCTURE_CORTEX_LEFT":
+            left_cortex = bm
+        elif bm.brain_structure == "CIFTI_STRUCTURE_CORTEX_RIGHT":
+            right_cortex = bm
+
+    if left_cortex is None or right_cortex is None:
+        raise RuntimeError("CIFTI is missing cortex information")
+
+    # calculate an average of the BOLD data, excluding the first 5 volumes
+    # as potential nonsteady states
+    data = img.get_fdata()[5:20].mean(axis=0)
```

Comment:
`get_fdata()` will load all timeseries and promote to `float64`. To ensure it stays memmapped:

```suggestion
    data = img.dataobj[5:20].mean(axis=0)
```